### DISCUSSION TITLE
Django 3.1 released with support for async views

### DISCUSSION THREAD

**Discussion by patrick91** [strawberry-graphql/strawberry#409] [General]
This is cool! https://www.djangoproject.com/weblog/2020/aug/04/django-31-released/

I think it is still not idea for strawberry as there's this caveat:

> Django’s ORM, cache layer, and other pieces of code that do long-running network calls do not yet support async access. We expect to add support for them in upcoming releases. Async views are ideal, however, if you are doing a lot of API or HTTP calls inside your view, you can now natively do all those HTTP calls in parallel to considerably speed up your view’s execution.

But can't wait to see how this evolves. I guess we should provide an async view as well soon :)

**Comment by lostb1t** [comment]
Having a async view with web socket support for subscriptions would be great. While having an async ORM is the next big step, the current state allows for example to create a event bus over subscriptions. Which is a common scenario.

I actually use Django heavily with graphQL and was waiting on this to drop so I might have a go at it pretty soon.

We would have to provide our own Websocket implementation. Not a biggie but good to note.

**Reply by patrick91** [reply]
yeah, I think not having the ORM working under async might be a big show stopper for most of people. But I'd be up for implementing subscription for django :)

**Comment by patrick91** [comment]
https://medium.com/@alex.oleshkevich/websockets-in-django-3-1-73de70c5c1ba
